### PR TITLE
AB2-666 Add Light Theme

### PR DESCRIPTION
**Pull request by moaz-cliqz**

### CODE REVIEW

File: mozilla-release/mobile/android/app/src/main/res/color/tint_edit_cancel_button.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (spacifici) [inline mozilla-release/mobile/android/app/src/main/res/color/tint_edit_cancel_button.xml]: File names have to be prefixed, i.e. `cliqz_tint_edit_cancel_button.xml`

File: mozilla-release/mobile/android/app/src/main/res/color/url_bar_site_security.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (spacifici) [inline mozilla-release/mobile/android/app/src/main/res/color/url_bar_site_security.xml]: Same here, please add the `cliqz_` or feature prefix

File: mozilla-release/mobile/android/app/src/photon/res/color/url_bar_title.xml [hunk 3]
```diff
@@ -7,10 +7,13 @@
...
 
     <!-- private browsing mode -->
-    <!-- Cliqz start o/
-    Cliqz does not have a different theme for private browsing mode
+    <!-- Cliqz start +->
     <item android:color="@color/photon_text_main_private" gecko:state_private="true" />
-    /o Cliqz end -->
+    <+- Cliqz end -->
+
+    <!-- Cliqz start -->
```
Comment (spacifici) [inline mozilla-release/mobile/android/app/src/photon/res/color/url_bar_title.xml]:
```xml
<!-- Cliqz start -->
<!-- item android:color="@color/photon_text_main_private" gecko:state_private="true" / -->

<item android:color="@android:color/white" gecko:light_theme="true"/>
<!-- Cliqz end -->

File: mozilla-release/mobile/android/app/src/photon/res/drawable/url_bar_bg.xml [hunk 4]
```diff
@@ -7,7 +7,13 @@
 <selector xmlns:android="http://schemas.android.com/apk/res/android" xmlns:gecko="http://schemas.android.com/apk/res-auto">
 
     <!-- Cliqz start -->
-    <!-- change the url background color on private mode -->
+    <!-- change the url background color on private mode, add light theme -->
+
+    <!-- private browsing mode, light theme -->
+    <item android:drawable="@color/light_theme_ghost_color" gecko:light_theme="true" gecko:state_private="true" />
```
Comment (spacifici) [inline mozilla-release/mobile/android/app/src/photon/res/drawable/url_bar_bg.xml]: Is it Ghosty color?

File: mozilla-release/mobile/android/app/src/photon/res/layout/browser_toolbar.xml [hunk 5]
```diff
@@ -67,13 +66,12 @@
         android:background="@drawable/url_bar_action_button"
         gecko:showLWTBackground="true"/>
 
-    <!-- remove height property -->
+    <!-- remove height, drawableTintList property -->
```
Comment (spacifici) [inline mozilla-release/mobile/android/app/src/photon/res/layout/browser_toolbar.xml]: If I may, why?

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/toolbar/PageActionLayout.java [hunk 6]
```diff
@@ -479,19 +495,23 @@ private void hidePreviousConfirmPrompt() {
...
         private final boolean mImportant;
+        /*Cliqz Start */
+        private boolean mUseTint;
 
         public PageAction(String id,
                           String title,
                           Drawable image,
                           OnPageActionClickListeners onPageActionClickListeners,
-                          boolean important) {
+                          boolean important, boolean useTint) {
```
Comment (spacifici) [inline mozilla-release/mobile/android/base/java/org/mozilla/gecko/toolbar/PageActionLayout.java]: Add a line break, please

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java [hunk 7]
```diff
@@ -235,7 +236,7 @@
                                    AntiPhishing.AntiPhishingCallback,
                                    AntiPhishingDialog.AntiPhishingDialogActionListener,
                                    /* Cliqz End */
-                                   OnboardingHelper.OnboardingListener {
+                                   OnboardingHelper.OnboardingListener, SharedPreferences.OnSharedPreferenceChangeListener {
```
Comment (spacifici) [inline mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java]: Add a line break, please.

Comment (spacifici) [reply mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java]: And it is not in the Cliqz block

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java [hunk 8]
```diff
@@ -4787,5 +4792,16 @@ private void refreshDrawables(){
     public boolean setRequestedOrientationForCurrentActivity(int requestedActivityInfoOrientation) {
         return super.setRequestedOrientationForCurrentActivity(requestedActivityInfoOrientation);
     }
+
+    private void setLightTheme() {
+        mBrowserToolbar.setLightTheme(mPreferenceManager.isLightThemeEnabled());
+    }
+    @Override
```
Comment (spacifici) [inline mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java]: Empy line

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java [hunk 9]
```diff
@@ -1015,6 +1019,7 @@ public void run() {
         if (AppConstants.Versions.feature24Plus) {
             maybeShowSetDefaultBrowserDialog(sharedPreferences, appContext);
         }
+        mPreferenceManager.registerOnSharedPreferenceChangeListener(this);
```
Comment (spacifici) [inline mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java]: The listener never get removed, it leaks the BrowserApp activity

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/home/TabMenuStripLayout.java [hunk 10]
```diff
@@ -37,19 +37,22 @@
...
+import org.mozilla.gecko.widget.themed.ThemedImageView;
 import org.mozilla.gecko.widget.themed.ThemedLinearLayout;
 
 /**
  * {@code TabMenuStripLayout} is the view that draws the {@code HomePager}
  * tabs that are displayed in {@code TabMenuStrip}.
  */
 class TabMenuStripLayout extends ThemedLinearLayout
-                         implements View.OnFocusChangeListener, Tabs.OnTabsChangedListener {
+                         implements View.OnFocusChangeListener, Tabs.OnTabsChangedListener, SharedPreferences.OnSharedPreferenceChangeListener {
```
Comment (spacifici) [inline mozilla-release/mobile/android/base/java/org/mozilla/gecko/home/TabMenuStripLayout.java]: Cliqz block

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/preferences/GeckoPreferences.java [hunk 11]
```diff
@@ -275,6 +273,9 @@
     public static final String PREFS_SHOW_CUSTOMIZE_TAB_VIEW = "pref.show.customize_tab.view";
     public static final String PREFS_SHOW_CUSTOMIZE_TAB_SNACKBAR = "pref.show.customize_tab.snackbar";
 
+    public static final String PREFS_BLUE_THEME = "pref.blue.theme";
+    private CheckBoxPreference showBackgroundPref;
```
Comment (spacifici) [inline mozilla-release/mobile/android/base/java/org/mozilla/gecko/preferences/GeckoPreferences.java]: `mShowBackgroundPref`

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/toolbar/BrowserToolbarPhoneBase.java [hunk 12]
```diff
@@ -108,62 +108,65 @@ public void onClick(View v) {
...
-        buttonsAnimator.start();
-    }
-
-    /**
-     * Returns the number of pixels the url bar translating edge
-     * needs to translate to the right to enter its editing mode state.
-     * A negative value means the edge must translate to the left.
-     */
+    /* Cliqz Start o/
+   @Override
```
Comment (spacifici) [inline mozilla-release/mobile/android/base/java/org/mozilla/gecko/toolbar/BrowserToolbarPhoneBase.java]: Indentation, there is a missing space.

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/toolbar/Ghosty.java [hunk 13]
```diff
@@ -40,7 +43,7 @@
     private float mTextDistance;
 
     private Rect mOutTextBound = new Rect();
-
+    private boolean mIsLightTheme;
```
Comment (spacifici) [inline mozilla-release/mobile/android/base/java/org/mozilla/gecko/toolbar/Ghosty.java]: Can you please add an empty line.

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java (hunk 9 above)
Comment (moaz-cliqz) [reply mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java]: done

**Comment by jayeshsolanki93** [comment]
Created a new PR with the changes #599

**Pull request closed without merging**